apps/client/src/app/shared-ui/raw-resource/raw-resource.component.ts

```diff
@@ -5,67 +5,53 @@ import {saveAs} from 'file-saver';
 import * as vkbeautify from 'vkbeautify';
 
 @Component({
-    selector: 'app-raw-resource',
-    templateUrl: './raw-resource.component.html',
-    styleUrls: ['./raw-resource.component.css']
+  selector: 'app-raw-resource',
+  templateUrl: './raw-resource.component.html',
+  styleUrls: ['./raw-resource.component.css']
 })
-export class RawResourceComponent implements OnInit, OnChanges {
-    @Input() shown?: boolean;
-    @Input() resource: Resource;
+export class RawResourceComponent {
+  @Input() shown?: boolean;
+  @Input() resource: Resource;
 
-    public xml: string;
+  constructor(private fhirService: FhirService) {
+  }
 
-    constructor(private fhirService: FhirService) {
+  public get baseFileName() {
+    if (this.resource  && this.resource.id) {
+      return this.resource.id;
     }
 
-    public get baseFileName() {
-        if (this.resource  && this.resource.id) {
-            return this.resource.id;
-        }
+    return 'resource';
+  }
 
-        return 'resource';
+  public downloadJson(event?) {
+    if (event) {
+      event.preventDefault();
+      event.stopPropagation();
     }
 
-    public downloadJson(event?) {
-        if (event) {
-            event.preventDefault();
-            event.stopPropagation();
-        }
+    const blob = new Blob([JSON.stringify(this.resource, null, '\t')], { type: 'application/json; charset=utf-8' });
+    saveAs(blob, this.baseFileName + '.json');
+  }
 
-        const blob = new Blob([JSON.stringify(this.resource, null, '\t')], { type: 'application/json; charset=utf-8' });
-        saveAs(blob, this.baseFileName + '.json');
+  public downloadXml(event?) {
+    if (event) {
+      event.preventDefault();
+      event.stopPropagation();
     }
 
-    public downloadXml(event?) {
-        if (event) {
-            event.preventDefault();
-            event.stopPropagation();
-        }
+    const resourceXml = this.fhirService.serialize(this.resource);
+    const blob = new Blob([resourceXml], { type: 'text/xml; charset=utf-8' });
+    saveAs(blob, this.baseFileName + '.xml');
+  }
 
-        const resourceXml = this.fhirService.serialize(this.resource);
-        const blob = new Blob([resourceXml], { type: 'text/xml; charset=utf-8' });
-        saveAs(blob, this.baseFileName + '.xml');
+  public get xml() {
+    if (!this.resource) {
+      return;
     }
 
-    private serialize() {
-        if (!this.resource) {
-            return;
-        }
+    const xml = this.fhirService.serialize(this.resource);
 
-        const xml = this.fhirService.serialize(this.resource);
-
-        if (xml) {
-            this.xml = vkbeautify.xml(xml);
-        } else {
-            this.xml = '';
-        }
-    }
-
-    ngOnInit() {
-        this.serialize();
-    }
-
-    ngOnChanges(changes: SimpleChanges): void {
-        this.serialize();
```

Comment:
Performance may be improved here by using an event with a debounce, so that `this.serialize()` is only called once every 500ms (or something like that).